emscripten.py

```diff
@@ -1232,80 +1232,41 @@ def finalize_output(metadata, post, funcs_js, sending, receiving, asm_setup, the
     first_in_asm = ''
     if settings['SPLIT_MEMORY']:
       if not settings['SAFE_SPLIT_MEMORY']:
-        first_in_asm += '''
-function get8(ptr) {
+        def make_get_set(name, coercer, shift):
+          access = 'HEAP{name}s[ptr >> SPLIT_MEMORY_BITS][(ptr & SPLIT_MEMORY_MASK) >> {shift}]'.format(name=name, shift=shift)
+          format_data = {
+            'name': name,
+            'coerced_value': coercer('value'),
+            'access': access,
+            'coerced_access': coercer(access),
+          }
+          getter = '''
+function get{name}(ptr) {{
```

Comment:
So this is interesting.

I'm using `'{a} {b}'.format(a=1, b=2)` instead of `'%s %s' % (1, 2)` here so I can give the parameters more explanatory names.
But then because we're writing a javascript function, the string contains `{`s, which we then have to escape as `{{`s, which was pretty non-obvious.
On top of that I pass the format dictionary with `**kwargs`, which I feel is one of the less obvious bits of python syntax.

I personally like the way this looks, but is this sufficiently non-obvious that `%s`s would be better?